mrmustard/lab/abstract/transformation.py

```diff
@@ -386,6 +386,26 @@ def __eq__(self, other):
         return True
 
     def __repr__(self):
+
+        class_name = self.__class__.__name__
+        modes = self.modes
+
+        parameters = {k: v for k, v in self.__dict__.items() if isinstance(v, Parameter)}
+        param_str_rep = [
+            f"{name}={repr(math.asnumpy(par.value))}" for name, par in parameters.items()
+        ]
+
+        params_str = ", ".join(sorted(param_str_rep))
+
+        return f"{class_name}({params_str}, modes = {modes})".replace("\n", "")
+
+    def __str__(self):
+
+        class_name = self.__class__.__name__
+        modes = self.modes
+        return f"<{class_name} object at {hex(id(self))} acting on modes {modes}>"
+
+    def _ipython_display_(self):
```

Comment:
For some reason, now when I create a circuit like

```
circuit = mm.Sgate(1,modes=[0,1])
circuit >>= mm.BSgate(np.pi/4,modes=[0,1])
```

and then have a cell in a jupyter notebook where I call `circuit`, it just prints an empty table.

Reply:
This method should be name `_repr_markdown`. This is how it looks now on a notebook for me

![image](https://user-images.githubusercontent.com/675763/175716681-e6f53579-65e7-440c-ad5b-6b81a5643a52.png)

Is it looking good? Can you try it out?

Reply:
It works for me now! Awesome!